URBNSwiftCarousel/Classes/URBNPresentationController.swift

```diff
@@ -0,0 +1,199 @@
+        transitionView.contentMode = .ScaleToFill
+        transitionView.image = sourceVC.imageForGalleryTransition()
+        
+        let scaleX = convertedStartingFrame.width / convertedEndingFrame.width
+        let scaleY = convertedStartingFrame.height / convertedEndingFrame.height
+        
+        let transform = CGAffineTransformMakeScale(scaleX, scaleY)
+        transitionView.transform = transform
+        transitionView.center = CGPointMake(CGRectGetMidX(convertedStartingFrame), CGRectGetMidY(convertedStartingFrame))
+        
+        containingView.addSubview(transitionView)
+        
+        sourceVC.willBeginGalleryTransitionWithImageView?(transitionView, isToVC: false)
+        destinationVC.willBeginGalleryTransitionWithImageView?(transitionView, isToVC: true)
+    }
+    
+    private func setFinalTransformForImage(sourceVC: URBNSwCarouselTransitioning, destinationVC: URBNSwCarouselTransitioning, containingView: UIView) {
+        let convertedStartingFrame = sourceVC.fromImageFrameForGalleryTransitionWithContainerView(containingView)
+        let convertedEndingFrame = destinationVC.toImageFrameForGalleryTransitionWithContainerView(containingView, sourceImageFrame: convertedStartingFrame)
+        
+        var center = CGPoint()
+        var transForm = CGAffineTransform()
+  
+        transForm = CGAffineTransformIdentity
+        center = CGPointMake(CGRectGetMidX(convertedEndingFrame), CGRectGetMidY(convertedEndingFrame))
+        
+        transitionView.center = center
+        transitionView.transform = transForm
+    }
+    
+    private func fireDelegatesAndReset(sourceVC: URBNSwCarouselTransitioning, destinationVC: URBNSwCarouselTransitioning, containingView: UIView) {
+        sourceVC.didEndGalleryTransitionWithImageView?(transitionView, isToVC: false)
+        destinationVC.didEndGalleryTransitionWithImageView?(transitionView, isToVC: true)
+        transitionView.removeFromSuperview()
+    }
+    
+    /**
+    *    This method syncs the cells between collection views by taking in the index path of the cell selected for zooming.  
+    *    That index path is sent to the destination collection view which scrolls to that index path.
+    */
+    private func synchronizeCollectionViews(sourceVC: URBNSwCarouselTransitioning, destinationVC: URBNSwCarouselTransitioning) {
+        guard let destSyncVC = destinationVC as? URBNSynchronizingDelegate, sourceSyncVC = sourceVC as? URBNSynchronizingDelegate, path = sourceSyncVC.sourceIndexPath(), cv = destSyncVC.toCollectionView() else { return }
+        
+        cv.scrollToItemAtIndexPath(path, atScrollPosition: .None, animated: false)
+        cv.reloadItemsAtIndexPaths([path])
+        if let cell = cv.cellForItemAtIndexPath(path) as? URBNCarouselZoomableCell {
+            destSyncVC.updateSourceSelectedCell?(cell)
+        }
+    }
+    
+    /**
+    *    Sometimes the presentingViewController will be the navigation controller of the presenting view controller.
```

Comment:
@ndis1  @codecaffeine  any thoughts here on this problem?  getting the cropped return rect of the image is working fine, but setting up the correct rects for the transition view isn't working.